include/swift/SIL/SILBasicBlock.h

```diff
@@ -352,17 +352,30 @@ public SwiftObjectHeader {
                                      ValueOwnershipKind kind,
                                      const ValueDecl *decl = nullptr);
 
+  SILPhiArgument *replacePhiArgument(unsigned i, SILType type,
```

Comment:
Would it make sense to have a single entry point for each of these (`replacePhiArgumnt`, `replacePhiArgumentAndReplaceAllUses`, `createPhiArgument`) with default arguments for `isEscaping` and `isReborrow`?

```
  SILPhiArgument *replacePhiArgument(unsigned i, SILType type,
                                      ValueOwnershipKind kind, bool isReborrow = false,
                                      bool isEscaping = false,
                                      const ValueDecl *decl = nullptr);
```

Reply:
@atrick had the same comment offline. It wasn't straightforward to do this because callers passing `decl` just got modified underneath to mean passing "true" to `isReborrow` because C++ does not consider this a type error. I'll make another pass to identify such callers and then change it to the new signature.

Reply:
Oh, C++.  Two other alternatives that come to mind:
- switch the order of arguments
- keep the new overloads with the additional arguments and call them from the original functions without the arguments

Reply:
Yeah, I think the trick is to always append to the default argument list to not run into problems like I did. But it's a bit unfortunate if the "previous last default argument" is likely to be passed infrequently..